servers/visual/visual_server_scene.cpp

```diff
@@ -2641,8 +2641,19 @@ void VisualServerScene::_prepare_scene(const Transform p_cam_transform, const Ca
 	RID *directional_light_ptr = &light_instance_cull_result[light_cull_count];
 	directional_light_count = 0;
 
-	// directional lights
-	{
+	if (bool(GLOBAL_GET("rendering/quality/shadows/update_every_2_frames"))) {
+		// Toggle between directional and point light shadow updating every frame.
```

Comment:
This technique is discouraged to be used every frame, as these string methods are not efficient.

If you look in the comment section at the top of `project_settings.h`, there are now two methods to only query this when project settings have changed.

From `VisualServerScene`, as it cannot receive signals, you probably want to cache this value, and call `ProjectSettings::has_changes()` and only call GLOBAL_GET if changes have occurred.

We can probably centralize this for all such project settings that need to be cached in e.g. `VisualServer`. If you are not sure how to do this, just remind me after such a change and I can put this in.